packages/mongodb-rag-core/src/Page.ts

```diff
@@ -54,26 +56,35 @@ export type PersistedPage = Page & {
   action: PageAction;
 };
 
+export type LoadPagesQuery = {
```

Comment:
why support both this and LoadPageArgs?

isn't LoadPagesArgs a subset of LoadPagesQuery?

Reply:
also maybe we could abstract this to a `Record<string, unknown>` here, and then in the makeMongoDbPageStore implementation use `Filter<PersistedPage>`. this way the Page type abstraction isn't tightly coupled to MongoDB.